packages/playwright-test/src/reporters/blob.ts

```diff
@@ -83,66 +112,103 @@ export class BlobReporter extends TeleReporterEmitter {
 
 export async function createMergedReport(config: FullConfigInternal, dir: string, reporterName?: string) {
   const shardFiles = await sortedShardFiles(dir);
-  const events = await mergeEvents(dir, shardFiles);
-
-  const defaultReporters: {[key in BuiltInReporter]: new(arg: any) => Reporter} = {
-    dot: DotReporter,
-    line: LineReporter,
-    list: ListReporter,
-    github: GitHubReporter,
-    json: JSONReporter,
-    junit: JUnitReporter,
-    null: EmptyReporter,
-    html: HtmlReporter,
-    blob: BlobReporter,
-  };
-  reporterName ??= 'list';
+  const resourceDir = path.join(dir, 'temp');
+  await fs.promises.mkdir(resourceDir, { recursive: true });
+  try {
+    const shardReports = await extractReports(dir, shardFiles, resourceDir);
+    const events = mergeEvents(shardReports);
+    patchAttachmentPaths(events, resourceDir);
 
-  const arg = config.config.reporter.find(([reporter, arg]) => reporter === reporterName)?.[1];
-  const options = {
-    ...arg,
-    configDir: process.cwd(),
-    outputFolder: dir
-  };
+    const defaultReporters: {[key in BuiltInReporter]: new(arg: any) => Reporter} = {
```

Comment:
Let's try to reuse `createReporter` function.

Reply:
I'll try to do that in a follow-up as agreed offline.